src/Idris/REPL.idr

```diff
@@ -949,6 +955,187 @@ process NOP
     = pure Done
 process ShowVersion
     = pure $ VersionIs  version
+process (ImportPackage package) = do
+  defs <- get Ctxt
+  let searchDirs = defs.options.dirs.extra_dirs
+  let Just packageDir = find
+        (\d => isInfixOf package (fromMaybe d (fileName d)))
+        searchDirs
+    | _ => pure (REPLError (pretty "Package not found in the known search directories"))
+  let packageDirPath = parse packageDir
+  tree <- coreLift $ explore packageDirPath
+  fentries <- coreLift $ toPaths (toRelative tree)
+  errs <- for fentries \entry => do
+    let entry' = dropExtension entry
+    let sp = forget $ split (== dirSeparator) entry'
+    let ns = concat $ intersperse "." sp
+    let ns' = mkNamespace ns
+    catch (do addImport (MkImport emptyFC False (nsAsModuleIdent ns') ns'); pure Nothing)
+          (\err => pure (Just err))
+  let errs' = catMaybes errs
+  res <- case errs' of
+    [] => pure (pretty "Done")
+    onePlus => pure $ vsep !(traverse display onePlus)
+  pure (Printed res)
+ where
+  toPaths : {root : _} -> Tree root -> IO (List String)
+  toPaths tree =
+    depthFirst (\x => map (toFilePath x ::) . force) tree (pure [])
+
+-- B or _ -> B or A -> B
+-- where A and B are spines of references.
+process (FuzzyTypeSearch expr) = do
+  let Just (neg, pos) = parseExpr expr
```

Comment:
Can we put this in a separate module? e.g. `Idris.REPL.FuzzySearch`

Reply:
Done. Also did a few refactorings around the new namespace. See the commit message

Reply:
Nice!